src/convolution/convolution_1D.jl

```diff
@@ -0,0 +1,1757 @@
+
+It's alright if you can't tell how the resulting function is created. In the following part we will go through each step of the convolution process, to understand better how they work."""
+
+# ╔═╡ f59990f8-ecba-4d16-8b68-f3d40b044d74
+md"""Use the slider below to change the amplitude of the filter function and the range on which it's defined."""
+
+# ╔═╡ 032130c4-686b-4db9-9753-9b0fe764f94e
+begin
+	nb_values = 8
+	amp_2_slider = @bind amp_2 PlutoUI.Slider(1:1:5, show_value=true, default=2)
+	nonzero_2_slider = @bind nonzero_2 PlutoUI.Slider(1:1:nb_values, show_value=true, default=3)
+	md"""Amplitude: $(amp_2_slider)
+		
+	Length: $(nonzero_2_slider)"""
+end
+
+# ╔═╡ 572bd8d5-65b0-462e-a143-811f4bfa875e
+md"""Ok, now we're ready to go! Move the slider around to see how the resulting point in the convoluted function was calculated."""
+
+# ╔═╡ bccfdd98-b425-4f8d-a58d-489134851ebd
+begin
+	k_slider = @bind k_conv PlutoUI.Slider(1:2*nb_values+1, show_value=true, default=8)
+	md"""Index: $(k_slider)"""
+end
+
+# ╔═╡ 0ccf60db-75b2-466d-935e-f39855bc36f7
+md"""## Generating Code """
+
+# ╔═╡ e4a17824-0e9f-442e-82bc-62b8d46e88e5
+md"""The following code sets up our initial figure. If you're interested in drawing graphs in computer science, you can take a look but it's not necessary to understand the mathematical concept."""
+
+# ╔═╡ 29752f8c-df55-4abc-868a-0e7404fad540
+md"""#### Helper Functions"""
+
+# ╔═╡ 8680db2e-3dda-4aff-a00e-130ac1f4486c
+function draw_point(x::Vector, y1::Vector, ax::Axis, marker::Symbol, color::ColorTypes.RGB{Float64}, offset::Int)
+		[scatter!(ax,repeat([x],y),(1:y).+offset,markersize=50,strokewidth=1,marker=marker, strokecolor=:white, color=color) for (x,y) in zip(x,y1)]
+end
+
+# ╔═╡ 50420349-f3f7-45bb-a3ab-0ac379a067bb
+function gaussian(x::Float64, μ, σ, range_min, range_max)
+    scaled_x = ((x - μ) / σ) * ((range_max - range_min) / 4) + ((range_max + range_min) / 2)
+    coeff = 1 / (σ * sqrt(2π))
+    exponent = -((scaled_x - μ)^2) / (2σ^2)
+    return coeff * exp(exponent)
+end
+
+# ╔═╡ 47cc0108-a86d-4e64-a013-8c4392e8987b
+let
+	f = Figure()
+	ax1 = CairoMakie.Axis(f[1, 1])
+	ax2 = CairoMakie.Axis(f[2, 1])	
+	l = length(z)
+
+	# First function
+	Random.seed!(250)
+	fz1 = rand(PinkGaussian(n))
+
+	# Second function
+	if fz2_function == 1
+		fz2 = ([gaussian(zi,50, 10, 0.0, 100.0) .* 80 for zi in collect(z)], [gaussian(zi,e, 10, 0.0, 100.0) .* 80 for zi in collect(z)])
+	else
+		fz2 = ((z.>=40 .&& z .<50) .* 3, (z.>=e-10 .&& z .<e+10) .* 3) 
+	end
+	
+	# Convoluted Function
+	fz3 = DSP.conv(fz1, reverse(fz2[1]))[end÷4+5:end÷4*3+6]
+
+	# Drawn moving functions 
+	fz2_draw = fz2[2]
+	fz3_draw = [ i < e ? fz3[i] : NaN for i in eachindex(fz3) ]
+
+	# Setting axes limits
+	xlims!(ax2, (0, 100))
+	ylims!(ax2, (0, 70))
+	
+	# Plotting
+	lines!(ax1, z, fz1, color = "red")
+	lines!(ax1, z, fz2_draw, color = "blue")
+	lines!(ax2, z, fz3, color="gray")
+	lines!(ax2, z, fz3_draw, color="purple")
+
+	f
```

Comment:
The logic here doesnt really work.

We want the top plot to show the 2 functions (lower left plot now), then we want a multiply sign, then we want the plot with currently the "plus" - indicating we did pointwise multiplications, the we want the result. Else it is quite hard

Also maybe the same scales would help to visualize everyhting better?